src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -138,40 +105,42 @@ BinaryCompactObject::BinaryCompactObject(
     time_dependence_ =
         std::make_unique<domain::creators::time_dependence::None<3>>();
   }
-  if (use_logarithmic_map_object_A_ and not excise_interior_A_) {
+  if (object_A_.use_logarithmic_map and not object_A_.is_excised()) {
     PARSE_ERROR(
         context,
-        "excise_interior_A must be true if use_logarithmic_map_object_A is "
-        "true; that is, using a logarithmically spaced radial grid in the part "
+        "Using a logarithmically spaced radial grid in the part "
         "of Layer 1 enveloping Object A requires excising the interior of "
         "Object A");
   }
-  if (use_logarithmic_map_object_B_ and not excise_interior_B_) {
+  if (object_B_.use_logarithmic_map and not object_B_.is_excised()) {
     PARSE_ERROR(
         context,
-        "excise_interior_B must be true if use_logarithmic_map_object_B is "
-        "true; that is, using a logarithmically spaced radial grid in the part "
+        "Using a logarithmically spaced radial grid in the part "
         "of Layer 1 enveloping Object B requires excising the interior of "
         "Object B");
   }
-  using domain::BoundaryConditions::is_none;
-  if (outer_boundary_condition_ != nullptr and
-      ((not excise_interior_A_ and not excise_interior_B_) and
-       (inner_boundary_condition_ == nullptr or
-        not is_none(inner_boundary_condition_)))) {
+  if (object_A_.is_excised() and
+      // NOLINTNEXTLINE(readability-implicit-bool-conversion)
```

Comment:
Apparently `xor` of bools returns an int in C++. On second though that makes sense I guess, since `!=` is just `xor` for bools. So I changed to that in the fixup :)